contracts/TopShotLocking.cdc

```diff
@@ -0,0 +1,114 @@
+import NonFungibleToken from 0xNFTADDRESS
+
+pub contract TopShotLocking {
+
+    // -----------------------------------------------------------------------
+    // TopShotLocking contract Events
+    // -----------------------------------------------------------------------
+
+    // Emitted when a moment is withdrawn from a Collection
+    pub event Locked(id: UInt64, expiryTimestamp: UFix64)
+    // Emitted when a moment is deposited into a Collection
+    pub event Unlocked(id: UInt64)
+
+    // Dictionary of locked NFTs
+    // nft resource uuid is the key
+    // locked until timestamp is the value
+    pub var lockedNFTs: {UInt64: UFix64} 
+
+    // Dictionary of NFTs overridden to be unlocked
+    pub var unlockableNFTs: {UInt64: Bool} // nft resource uuid is the key
+
+    // isLocked Returns a boolean indicating if an nft exists in the lockedNFTs dictionary
+    //
+    // Parameters: nftRef: A reference to the NFT resource
+    //
+    // Returns: true if NFT is locked
+    pub fun isLocked(nftRef: &NonFungibleToken.NFT): Bool {
+        return self.lockedNFTs.containsKey(nftRef.uuid)
+    }
+
+    // lockNFT Takes an NFT resource and adds its unique identifier to the lockedNFTs dictionary
+    //
+    // Parameters: nft: NFT resource
+    //
+    // Returns: the NFT resource
+    pub fun lockNFT(nft: @NonFungibleToken.NFT, expiryTimestamp: UFix64): @NonFungibleToken.NFT {
+        let TopShotNFTType: Type = CompositeType("A.0xTOPSHOTADDRESS.TopShot.NFT")!
+        if !nft.isInstance(TopShotNFTType) {
+            panic("NFT is not a TopShot NFT")
+        }
+
+        let id = nft.uuid
+        if self.lockedNFTs.containsKey(id) {
+            panic("NFT is already locked")
+        }
```

Comment:
probably don't need to panic here. You can just skip the locking and event if it is already locked